Generated docs fail to render

### Zig Version

0.15.0-dev.1218+bc8e1a74c

### Steps to Reproduce and Observed Behavior

My build script has the following setup for generating docs:
```zig
const docs_exe = b.addExecutable(.{
    .name = "zcs",
    .root_module = b.createModule(.{
        .root_source_file = b.path("src/main.zig"),
        .target = target,
        .optimize = optimize,
    }),
    .use_llvm = false,
});
const docs = docs_exe.getEmittedDocs();
const install_docs = b.addInstallDirectory(.{
    .source_dir = docs,
    .install_dir = .prefix,
    .install_subdir = "docs",
});
const docs_step = b.step("docs", "Build the docs");
docs_step.dependOn(&install_docs.step);
```

I then generate them like this:
```sh
zig build docs
```

And serve them like this:
```
python3 -m http.server --directory zig-out/docs
```

When I then try to view them in a browser, I get the following error:
```
ubsan_rt/c.zig: expected newline at end of file
panic: unable to unpack tar: EndOfStream
```

I was able to repro this with a minimal main file that just defines the following:
```zig
const std = @import("std");

pub fn main() !void {
    // Prints to stderr, ignoring potential errors.
    std.debug.print("All your {s} are belong to us.\n", .{"codebase"});
}

pub fn foo() void {}
```

### Expected Behavior

I expected to see the rendered docs.

Applies to the zig std lib docs at https://ziglang.org/documentation/master/std/ as well
```
Errors

std/tz.zig: expected newline at end of file
panic: unable to unpack tar: EndOfStream
```

--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -4880,6 +4880,14 @@ fn docsCopyFallible(comp: *Compilation) anyerror!void {
         try seen_table.ensureUnusedCapacity(comp.gpa, deps.len);
         for (deps) |dep| seen_table.putAssumeCapacity(dep, dep.fully_qualified_name);
     }
+
+    tar_file_writer.end() catch |err| {
+        return comp.lockAndSetMiscFailure(
+            .docs_copy,
+            "unable to write '{f}/sources.tar': {t}",
+            .{ docs_path, err },
+        );
+    };
 }
 
 fn docsCopyModule(

